packages/components/Separator/src/Separator.tsx

```diff
@@ -1,23 +1,31 @@
+/** @jsx withSlots */
 import { View } from 'react-native';
-import { separatorName, ISeparatorType } from './Separator.types';
-import { compose } from '@uifabricshared/foundation-compose';
-import { filterViewProps } from '@fluentui-react-native/adapters';
-import { separatorTokenProcessor } from './Separator.helpers';
-import { settings } from './Separator.settings';
-import { getPaletteFromTheme } from '@fluentui-react-native/tokens';
+import { SeparatorProps, SeparatorTokens, SeparatorType } from './Separator.types';
+import { compose, UseSlots, buildProps, withSlots, mergeProps } from '@fluentui-react-native/framework';
 
-export const Separator = compose<ISeparatorType>({
+export const separatorName = 'Separator';
+const propMask = { vertical: undefined };
+
+export const Separator = compose<SeparatorType>({
   displayName: separatorName,
-  settings: settings,
-  slots: {
-    root: {
-      slotType: View,
-      filter: filterViewProps
-    }
+  tokens: [{ separatorWidth: 1 }, separatorName],
+  tokensThatAreAlsoProps: ['vertical'],
```

Comment:
what does tokensThatAreAlsoProps do under the hood?

Reply:
The main function is that it controls which props are copied into tokens before styles are prepared. So in this case the tokens will come from the theme + defaults, including a value for vertical. But if vertical is specified in props it will be copied into tokens before being handed to the slotProps functions.

The more subtle function is that the buildProps helper will actually filter the keys based on the intersection of this and its own list. So if you have `buildProps(tokens => { /* build props here */ }, ['vertical', 'horizontal'])` and only vertical is listed, the keys will be reduced to only be vertical. This is because the tokens before props are applied are already keyed on the theme, so only the ones that may be changed via props need to be used as memo keys.